javascript/packages/orchestrator/src/providers/podman/resources/node.resource.ts

```diff
@@ -0,0 +1,177 @@
+import { getRandomPort, makeDir } from "@zombienet/utils";
+import { genCmd, genCumulusCollatorCmd } from "../../../cmdGenerator";
+import {
+  P2P_PORT,
+  PROMETHEUS_PORT,
+  RPC_HTTP_PORT,
+  RPC_WS_PORT,
+} from "../../../constants";
+import { Node } from "../../../types";
+import { Client } from "../../client";
+
+export class NodeResource {
+  private readonly configPath: string;
+  private readonly dataPath: string;
+  private readonly relayDataPath: string;
+
+  constructor(
+    client: Client,
+    protected readonly namespace: string,
+    protected readonly nodeSetupConfig: Node,
+  ) {
+    const nodeName = nodeSetupConfig.name;
+    this.configPath = `${client.tmpDir}/${nodeName}/cfg`;
+    this.dataPath = `${client.tmpDir}/${nodeName}/data`;
+    this.relayDataPath = `${client.tmpDir}/${nodeName}/relay-data`;
+  }
+
+  public async generateSpec() {
+    const volumes = await this.generateVolumes();
+    const volumeMounts = this.generateVolumesMounts();
+    const containersPorts = await this.generateContainersPorts();
+    const containers = await this.generateContainers(
+      volumeMounts,
+      containersPorts,
+    );
+    const podManifest = this.generatePodSpec(containers, volumes);
+
+    return podManifest;
+  }
+
+  private async createVolumeDirectories() {
+    await makeDir(this.configPath, true);
+    await makeDir(this.dataPath, true);
+    await makeDir(this.relayDataPath, true);
+  }
+
+  private async generateVolumes() {
+    await this.createVolumeDirectories();
+
+    const configVolume = {
+      name: "tmp-cfg",
+      hostPath: { type: "Directory", path: this.configPath },
+    };
+    const dataVolume = {
+      name: "tmp-data",
+      hostPath: { type: "Directory", path: this.dataPath },
+    };
+    const relayDataVolume = {
+      name: "tmp-relay-data",
+      hostPath: { type: "Directory", path: this.relayDataPath },
+    };
+
+    return [configVolume, dataVolume, relayDataVolume];
+  }
+
+  private generateVolumesMounts() {
+    const configVolumeMount = {
+      name: "tmp-cfg",
+      mountPath: "/cfg:U",
+      readOnly: false,
+    };
+    const dataVolumeMount = {
+      name: "tmp-data",
+      mountPath: "/data:U",
+      readOnly: false,
+    };
+    const relayDataVolumeMount = {
+      name: "tmp-relay-data",
+      mountPath: "/relay-data:U",
+      readOnly: false,
+    };
+
+    return [configVolumeMount, dataVolumeMount, relayDataVolumeMount];
+  }
+
+  private async portFromNodeSetupConfigOrDefault(
+    portProperty: keyof NonNullable<Node["externalPorts"]>,
+  ) {
+    const { externalPorts } = this.nodeSetupConfig;
+
+    if (externalPorts && portProperty in externalPorts) {
+      return externalPorts[portProperty];
+    }
```

Comment:
Im not sure if what I am proposing here is correct but Isn't this the same?
```suggestion
    if (externalPorts?.portProperty)
      return externalPorts[portProperty];
```

Reply:
You are right, it was a remainder when I was trying to find the correct type for the function argument

Reply:
I mean its just a proposal . if you think its a good idea go for it...same goes for most of the comments